src/Qowaiv/Financial/InternationalBankAccountNumber.Bban.cs

```diff
@@ -0,0 +1,47 @@
+﻿#pragma warning disable S1210
+// "Equals" and the comparison operators should be overridden when implementing "IComparable"
+// See README.md => Sortable
+
+#pragma warning disable S2328
+// "GetHashCode" should not reference mutable fields
+// See README.md => Hashing
+
+using Qowaiv.Globalization;
+using Qowaiv.Text;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Qowaiv.Financial
+{
+    public partial struct InternationalBankAccountNumber
+    {
+        /// <summary>Constructs a <see cref="Regex"/> based on its BBAN pattern.</summary>
+        private static KeyValuePair<Country, Regex> Bban(Country country, string bban, int? checksum = default)
+        {
+            var blocks = bban.Split(',').Select(b => b.Buffer());
+            var pattern = CharBuffer.Empty(64)
+                .Add('^')
+                .Add(country.IsoAlpha2Code)
+                .Add(checksum.HasValue ? checksum.Value.ToString("00") : "[0-9][0-9]");
```

Comment:
Why not use [0-9]{2} instead of [0-9][0-9]?

Reply:
@LauraXiulan also possible. I think this is easier to read. But it is matter of taste.